src/syft.js

```diff
@@ -7,159 +7,79 @@ import {
   WEBRTC_PEER_LEFT,
   WEBRTC_PEER_OPTIONS
 } from './_constants';
-import { NOT_ENOUGH_ARGS, NO_PLAN, PLAN_ALREADY_COMPLETED } from './_errors';
 
 import EventObserver from './events';
 import Logger from './logger';
 import Socket from './sockets';
 import WebRTCClient from './webrtc';
 import { protobuf, unserialize } from './protobuf';
+import GridAPIClient from './grid_api_client';
+import Job from './job';
+import SyftModel from './syft_model';
 
 export default class Syft {
   /* ----- CONSTRUCTOR ----- */
-  constructor({ url, verbose, workerId, scopeId, protocolId, peerConfig }) {
-    // My worker ID
-    this.workerId = workerId || null;
-
-    // The assigned scope ID
-    this.scopeId = scopeId || null;
-
-    // The chosen protocol we are working on
-    this.protocolId = protocolId || null;
-
-    // Our role in the protocol
-    this.role = null;
-
-    // The participants we will be working with (only for scope creators)
-    this.participants = [];
-
-    // The protocol we will are participating in
-    this.protocol = null;
+  constructor({ url, verbose, authToken, peerConfig }) {
+    // For creating verbose logging should the worker desire
+    this.logger = new Logger('syft.js', verbose);
 
-    // The plan we have been assigned
-    this.plan = null;
+    this.gridClient = new GridAPIClient({ url, logger: this.logger });
 
-    // The index of the last plan operation we weren't able to complete (this defaults to 0 until we've started to execute)
-    this.lastUnfinishedOperation = 0;
+    // models registry
+    this.models = new Map();
 
-    // All the tensors we've either computed ourselves or captured from Grid or other peers
-    this.objects = {};
+    // objects registry
+    this.objects = new Map();
 
     // For creating event listeners
     this.observer = new EventObserver();
 
-    // For creating verbose logging should the worker desire
-    this.logger = new Logger('syft.js', verbose);
-
-    // Create a socket connection at this.socket
-    this.socket = null;
-    this.createSocketConnection(url);
-
-    // The WebRTC client used for P2P communication
-    this.rtc = null;
-    this.createWebRTCClient(peerConfig);
+    this.id = null;
+    this.peerConfig = peerConfig;
+    this.authToken = authToken;
   }
 
-  /* ----- FUNCTIONALITY ----- */
-
-  // Get the protocol and plan assignment from the Grid
-  getProtocol(protocol) {
-    if (!protocol && !this.protocolId) return null;
-    if (protocol && !this.protocolId) this.protocolId = protocol;
+  async newJob({ modelId, modelVersion }) {
+    const authResponse = await this.gridClient.authenticate(this.authToken);
```

Comment:
We should only call this (and the line below) if we DO NOT have a value for `this.worker_id` yet.